vllm/model_executor/layers/quantization/fp8fnuz.py

```diff
@@ -0,0 +1,262 @@
+import os.path
+from typing import List, Optional, Tuple, Union
+
+import pandas as pd
+import torch
+from torch.nn import Module
+from torch.nn.parameter import Parameter
+
+import vllm.envs as env
+from vllm import _custom_ops as ops
+from vllm.logger import init_logger
+from vllm.model_executor.layers.linear import LinearBase, LinearMethodBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig)
+from vllm.model_executor.utils import set_weight_attrs
+from vllm.utils import is_hip
+
+logger = init_logger(__name__)
+
+
+class Fp8FnuzConfig(QuantizationConfig):
```

Comment:
Can we call this `QuarkConfig` instead?

Reply:
I'm a bit worry about these duplications. We now have
- `Fp8` -> `float8_e4m3fn`
- `Fp8Fnuz` -> `float8_e4m3fnuz`
- `FbGemmFp8` -> ...

Let's not talk about fbgemm first. For the first two it should be possible to improve the current `Fp8` to support `float8_e4m3fnuz` instead of duplicating another Fp8 method.